cpp/cugraph_module/algorithms/hits.cu

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2016-2022 Memgraph Ltd. [https://memgraph.com]
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mg_cugraph_utility.hpp"
+
+namespace {
+using vertex_t = int64_t;
+using edge_t = int64_t;
+using weight_t = double;
+using result_t = double;
+
+constexpr char const *kProcedureHITS = "get";
+
+constexpr char const *kArgumentTolerance = "tolerance";
+constexpr char const *kArgumentMaxIterations = "max_iterations";
+constexpr char const *kArgumentNormalize = "normalize";
+constexpr char const *kArgumentDirected = "directed";
+
+constexpr char const *kResultFieldNode = "node";
+constexpr char const *kResultFieldHubScore = "hubs";
+constexpr char const *kResultFieldAuthoritiesScore = "authorities";
+
+void InsertHITSRecord(mgp_graph *graph, mgp_result *result, mgp_memory *memory, const std::uint64_t node_id,
+                      double hubs, double authorities) {
+  auto *record = mgp::result_new_record(result);
+  mg_utility::InsertNodeValueResult(graph, record, kResultFieldNode, node_id, memory);
+  mg_utility::InsertDoubleValueResult(record, kResultFieldHubScore, hubs, memory);
+  mg_utility::InsertDoubleValueResult(record, kResultFieldAuthoritiesScore, authorities, memory);
+}
+
+void HITSProc(mgp_list *args, mgp_graph *graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    auto tolerance = mgp::value_get_double(mgp::list_at(args, 0));
+    auto max_iterations = mgp::value_get_int(mgp::list_at(args, 1));
+    auto normalize = mgp::value_get_bool(mgp::list_at(args, 2));
+    auto directed = mgp::value_get_bool(mgp::list_at(args, 3));
+
+    raft::handle_t handle{};
+    auto stream = handle.get_stream();
+
+    // Works with unweighted graph
+    auto graph_type = directed ? mg_graph::GraphType::kDirectedGraph: mg_graph::GraphType::kUndirectedGraph;
+    auto mg_graph = mg_utility::GetGraphView(graph, result, memory, graph_type);
+    if (mg_graph->Empty()) return;
```

Comment:
We can put this part here before following part:
```
raft::handle_t handle{};
auto stream = handle.get_stream();
```

Reply:
No problems, sounds like a good thing to do